packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -277,6 +281,16 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
         Object.keys(editRowsState).length > 0);
 
     const editCellState = editRowsState[rowId]?.[column.field] ?? null;
+    let cellIsNotVisible = false;
```

Comment:
I haven't tested live yet, but could you summarize briefly what rows/columns we render when the focused cell is outside the range?